packages/search/src/features/fhir/fhir-utils.ts

```diff
@@ -91,22 +96,28 @@ export function findEntry<T extends Resource = Resource>(
 }
 
 export async function addEventLocation(
+  bundle: SavedBundle,
   body: IBirthCompositionBody | IDeathCompositionBody,
-  code: CompositionSectionCode,
-  composition: SavedComposition
+  code: Extract<
+    CompositionSectionCode,
+    'birth-encounter' | 'death-encounter' | 'marriage-encounter'
+  >
 ) {
-  let data
-  let location: fhir.Location | undefined
+  const composition = getComposition(bundle)
+  let location: SavedLocation | undefined
 
   const encounterSection = findCompositionSection(code, composition)
   if (encounterSection && encounterSection.entry) {
-    data = await getFromFhir(
-      `/Encounter/${encounterSection.entry[0].reference}`
+    const encounter = getResourceFromBundleById<Encounter>(
+      bundle,
+      resourceIdentifierToUUID(encounterSection.entry[0].reference)
     )
 
-    if (data && data.location && data.location[0].location) {
-      location = await getFromFhir(`/${data.location[0].location.reference}`)
-    }
+    if (encounter && encounter.location)
```

Comment:
Is there any possibility of the encounter not being there? If not we can remove the check
```suggestion
    if (encounter.location)
```

Reply:
For an incomplete declaration, the encounter could be not present in the bundle. In that case, it would be useful, no?

Reply:
For incomplete declaration, if the encounter is not there, `getFromFhir` will throw an error, I think,  instead of returning `undefined` encounter 🤔

Reply:
@tahmidrahman-dsi you are right, this function will actually throw an error if the resource is not found in the bundle. @Nil20 we probably want to use `findResourceFromBundleById` in this case.

Reply:
For an incomplete declaration, the `encounterSection` would be undefined. There's a check prior to this line if the `encounterSection` exists. But if the section `birth-encounter` exists, then the encounter and encounter.location would exist as well. But since we don't want to throw any error here for indexing, it should be safer to use `findResourceFromBundleById`

Reply:
you're right about that, if there is a encounter section then it's safe to assume that the resource is there in the bundle